src/System.Windows.Forms/src/System/Windows/Forms/CheckBox.CheckBoxAccessibleObject.cs

```diff
@@ -48,33 +47,58 @@ public override AccessibleRole Role
                     {
                         return role;
                     }
+
                     return AccessibleRole.CheckButton;
                 }
             }
 
             public override AccessibleStates State
-            {
-                get
+                => OwningCheckBox.CheckState switch
                 {
-                    switch (((CheckBox)Owner).CheckState)
-                    {
-                        case CheckState.Checked:
-                            return AccessibleStates.Checked | base.State;
-                        case CheckState.Indeterminate:
-                            return AccessibleStates.Indeterminate | base.State;
-                    }
+                    CheckState.Checked => AccessibleStates.Checked | base.State,
+                    CheckState.Indeterminate => AccessibleStates.Indeterminate | base.State,
+                    _ => base.State
+                };
 
-                    return base.State;
-                }
-            }
+            internal override UiaCore.ToggleState ToggleState
+                => OwningCheckBox.Checked ? UiaCore.ToggleState.On : UiaCore.ToggleState.Off;
+
+            internal override bool IsPatternSupported(UiaCore.UIA patternId) =>
+                patternId switch
+                {
+                    var p when
+                        p == UiaCore.UIA.TogglePatternId ||
+                        p == UiaCore.UIA.LegacyIAccessiblePatternId => true,
+                    _ => base.IsPatternSupported(patternId)
+                };
+
+            internal override object GetPropertyValue(UiaCore.UIA propertyID)
+                => propertyID switch
+                {
+                    UiaCore.UIA.NamePropertyId
+                        => Name,
+                    UiaCore.UIA.AutomationIdPropertyId
+                        => OwningCheckBox.IsHandleCreated ? OwningCheckBox.Name : String.Empty,
```

Comment:
❗ We only use C# type aliases
```suggestion
                        => OwningCheckBox.IsHandleCreated ? OwningCheckBox.Name : string.Empty,
```

Reply:
Thanks, fixed.